entity-framework/core/what-is-new/ef-core-8.0/whatsnew.md

```diff
@@ -1127,3 +1127,468 @@ In addition to the enhancements described above, EF8 Preview 2 also includes som
+> ```
+
+Let's try a different kind of `Contains` query. In this case, we'll look for a value of the parameter collection in the column. For example, any pub that stocks Heineken:
+
+<!--
+        var beer = "Heineken";
+        var pubsWithHeineken = await context.Pubs
+            .Where(e => e.Beers.Contains(beer))
+            .Select(e => e.Name)
+            .ToListAsync();
+-->
+[!code-csharp[PubsWithHeineken](../../../../samples/core/Miscellaneous/NewInEFCore8/PrimitiveCollectionsSample.cs?name=PubsWithHeineken)]
+
+This translates to the following on SQL Server:
+
+```sql
+SELECT [p].[Name]
+FROM [Pubs] AS [p]
+WHERE EXISTS (
+    SELECT 1
+    FROM OpenJson([p].[Beers]) AS [b]
+    WHERE [b].[value] = @__beer_0)
+```
+
+`OpenJson` is now used to to extract values from JSON column so that each value can be matched to the passed parameter.
+
+We can combine the use of `OpenJson` on the parameter with `OpenJson` on the column. For example, to find pubs that stock any one of a variety of lagers:
+
+<!--
+        var beers = new[] { "Carling", "Heineken", "Stella Artois", "Carlsberg" };
+        var pubsWithLager = await context.Pubs
+            .Where(e => beers.Any(b => e.Beers.Contains(b)))
+            .Select(e => e.Name)
+            .ToListAsync();
+-->
+[!code-csharp[PubsWithLager](../../../../samples/core/Miscellaneous/NewInEFCore8/PrimitiveCollectionsSample.cs?name=PubsWithLager)]
+
+This translates to the following on SQL Server:
+
+```sql
+SELECT [p].[Name]
+FROM [Pubs] AS [p]
+WHERE EXISTS (
+    SELECT 1
+    FROM OpenJson(@__beers_0) AS [b]
+    WHERE EXISTS (
+        SELECT 1
+        FROM OpenJson([p].[Beers]) AS [b0]
+        WHERE [b0].[value] = [b].[value] OR ([b0].[value] IS NULL AND [b].[value] IS NULL)))
+```
+
+The `@__beers_0` parameter value here is `["Carling","Heineken","Stella Artois","Carlsberg"]`.
+
+Let's look at a query that makes use of the column containing a collection of dates. For example, to find pubs visited this year:
+
+<!--
+        var thisYear = DateTime.Now.Year;
+        var pubsVisitedThisYear = await context.Pubs
+            .Where(e => e.DaysVisited.Any(v => v.Year == thisYear))
+            .Select(e => e.Name)
+            .ToListAsync();
+-->
+[!code-csharp[PubsVisitedThisYear](../../../../samples/core/Miscellaneous/NewInEFCore8/PrimitiveCollectionsSample.cs?name=PubsVisitedThisYear)]
+
+This translates to the following on SQL Server:
+
+```sql
+SELECT [p].[Name]
+FROM [Pubs] AS [p]
+WHERE EXISTS (
+    SELECT 1
+    FROM OpenJson([p].[DaysVisited]) AS [d]
+    WHERE DATEPART(year, CAST([d].[value] AS date)) = @__thisYear_0)
+```
+
+Notice that the query makes use of the date-specific function `DATEPART` here because EF _knows that the primitive collection contains dates_. It might not seem it, but this is actually really important. Because EF knows what's in the collection, it can generate appropriate SQL to use the typed values with parameters, functions, other columns etc.
```

Comment:
```suggestion
Notice that the query makes use of the date-specific function `DATEPART` here because EF _knows that the primitive collection contains dates_. It might not seem like it, but this is actually really important. Because EF knows what's in the collection, it can generate appropriate SQL to use the typed values with parameters, functions, other columns etc.
```